src/providers/oauth-provider.js

```diff
@@ -93,6 +100,44 @@ function OAuthProvider() {
       }
 
       /**
+       * Requests a authorization for an application based on clientId, scope and state
+       *
+       * @param {string} clientId - Application `clientId`
+       * @param {string} scope - Scope(s) defined for the application
+       * @param {string} state - Randomly generated `state` string
+       * @return {promise} A response promise.
+       */
+
+      authorize(clientId, scope, state) {
+        // Check if `clientId` is defined.
+        if (!clientId) {
+          throw new Error('You must provide an application `clientId`');
+        }
+
+        const data = {
+          client_id: clientId,
+          response_type: 'code'
+        };
+
+        if (scope) {
```

Comment:
_query-string_ handles `undefined` values by not appending it to the stringified component. I would suggest to assign it to data directly such as:
```js
data = angular.extend({
  scope,
  state
}, data);
```